### ISSUE TITLE
Is there any difference between the reimplement of loss function and the formula in the paper?

### ISSUE THREAD

**Issue by walsvid** [eragonruan/text-detection-ctpn#70]
I have a problem with the implementation of the loss function, the formula of the loss function in the paper is as follows:
![image](https://user-images.githubusercontent.com/17134911/34369760-b221f82e-eaf9-11e7-8ea9-6ec974565364.png)
The 2nd term in the formula calculate `v`, the paper mentioned that `v={vc, vh}`
![image](https://user-images.githubusercontent.com/17134911/34369840-4781faae-eafa-11e7-9098-bc198b570459.png)
However, according to `lib/networks/network.py` the implement of loss function seems like using `rpn_bbox_pred` which contains 4 value : `tx, ty, tw, th`.
I would like to know if the current implementation has this problem, and if I want to use only `vc` and `vh`, what is the value order(`[x,y,w,h]` or `[x,w,y,h]` or others) in `rpn_bbox_pred`?

**Comment by eragonruan** [comment]
@walsvid good question！ in [/lib/networks/network.py](https://github.com/eragonruan/text-detection-ctpn/blob/5d091787c9ba2e0c5da919341a9fe11117cceab2/lib/networks/network.py#L399) when claculated the bbox_loss, the bbox_loss is multiplied by a weight matrix, rpn_bbox_inside_weights. And this matrix is set to [0,1,0,1](https://github.com/eragonruan/text-detection-ctpn/blob/5d091787c9ba2e0c5da919341a9fe11117cceab2/ctpn/text.yml#L29), so only the ty and th loss are calculated. Cause this repo is modify from Faster-RCNN, hence I implemented this part in this way for convenience

**Comment by walsvid** [comment]
@eragonruan I noticed the value `BBOX_INSIDE_WEIGHTS ` in configuration file [text.yml](https://github.com/eragonruan/text-detection-ctpn/blob/5d091787c9ba2e0c5da919341a9fe11117cceab2/ctpn/text.yml#L29). 
However, in [network.py](https://github.com/eragonruan/text-detection-ctpn/blob/5d091787c9ba2e0c5da919341a9fe11117cceab2/lib/networks/network.py#L392), `rpn_bbox_inside_weights` was calculated in layer named `rpn-data`, this layer actually call the function in [anchor_target_layer_tf.py](https://github.com/eragonruan/text-detection-ctpn/blob/5d091787c9ba2e0c5da919341a9fe11117cceab2/lib/rpn_msr/anchor_target_layer_tf.py). 

What confused me is a assign statement in this [line](https://github.com/eragonruan/text-detection-ctpn/blob/5d091787c9ba2e0c5da919341a9fe11117cceab2/lib/rpn_msr/anchor_target_layer_tf.py#L208) of `anchor_target_layer_tf`, it use `cfg.TRAIN.RPN_BBOX_INSIDE_WEIGHTS` set to `bbox_inside_weights` rather than `BBOX_INSIDE_WEIGHTS`, the value of `RPN_BBOX_INSIDE_WEIGHTS` seems to [1,1,1,1](https://github.com/eragonruan/text-detection-ctpn/blob/5d091787c9ba2e0c5da919341a9fe11117cceab2/ctpn/text.yml#L30), `BBOX_INSIDE_WEIGHTS` is not be used. I wonder is it correct?

**Comment by eragonruan** [comment]
@walsvid Thanks for the noticing. It seems that the matrix did not work. I will check it soon.

**Comment by walsvid** [comment]
Thank you for your reply, I have opened a PR. If it can fix the bug, I hope you can merge the code. If not, please do let me know whether there are any more problems with this.

**Comment by eragonruan** [comment]
I'll check it soon, since the model have to be retrained to verify whether if it affects the results , by the way, maybe rewrite this part according the the origin paper is the best way.
